crates/dojo/lang/src/attribute_macros/patches/library.patch.cairo

```diff
@@ -0,0 +1,19 @@
+#[starknet::contract]
+pub mod $name$ {
+    use dojo::contract::ILibrary;
+    use dojo::meta::IDeployedResource;
+
+    #[abi(embed_v0)]
+    pub impl $name$__LibraryImpl of ILibrary<ContractState> {}
+
+
+    // TODO: rename impl ??
+    #[abi(embed_v0)]
+    pub impl $name$__DeployedContractImpl of IDeployedResource<ContractState> {
+        fn dojo_name(self: @ContractState) -> ByteArray {
+            "$name$"
+        }
+    }
```

Comment:
We may not be able to use this at all due to the library calls being unsafe. Can be removed, or kept for debugging purposes.